tests/Doctrine/Migrations/Tests/Configuration/Connection/ConnectionLoaderTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Tests\Configuration\Connection;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\Migrations\Configuration\Connection\ConfigurationFile;
+use Doctrine\Migrations\Configuration\Connection\Exception\FileNotFound;
+use Doctrine\Migrations\Configuration\Connection\Exception\InvalidConfiguration;
+use Doctrine\Migrations\Configuration\Connection\ExistingConnection;
+use PHPUnit\Framework\TestCase;
+
+final class ConnectionLoaderTest extends TestCase
+{
+    public function testExistingConnectionLoader() : void
+    {
+        $conn   = $this->createMock(Connection::class);
```

Comment:
If we use a recent enough version of phpunit, we should use `createStub` instead here.

https://twitter.com/s_bergmann/status/1215163783229988864

Reply:
the codebase is full of createMock for stubs. createStub appears 0 times... but will do it here

Reply:
... PHPStan does not detect it as `Connection`.... will keep `createMock` for now.

Reply:
> but will do it here

Ok great!

> ... PHPStan does not detect it as Connection.... will keep createMock for now.

Oh bummer, no idea why it doesn't work…

Reply:
> > ... PHPStan does not detect it as Connection.... will keep createMock for now.
> 
> Oh bummer, no idea why it doesn't work…

phpstan-phpunit may not know about `createStub`, you can create an issue there to add support for it.

Reply:
Looks like it does know about it, but we are not up to date: https://github.com/phpstan/phpstan-phpunit/commit/8f609f69147772b8b59b0a3bf5e17009112de412

Reply:
Ok, good to know